src/test/java/com/gradle/develocity/bamboo/MavenExtensionsTest.java

```diff
@@ -0,0 +1,117 @@
+    void hasExtensionReturnsFalseWhenCoordinatesArgumentIsNull() throws IOException {
+        FileUtils.writeLines(extensionsXml, Collections.singleton(generateExtensionsXml(SOME_EXTENSION_COORDINATES)));
+
+        MavenExtensions mavenExtensions = MavenExtensions.fromFile(extensionsXml);
+
+        assertFalse(mavenExtensions.hasExtension(null));
+    }
+
+    @Test
+    void hasExtensionReturnsFalseWhenCoordinatesArgumentHasInvalidCharactersForXpathExpression() throws IOException {
+        MavenCoordinates coordsWithInvalidCharacters = new MavenCoordinates("com.example", "\"");
+        FileUtils.writeLines(extensionsXml, Collections.singleton(generateExtensionsXml(SOME_EXTENSION_COORDINATES)));
+
+        MavenExtensions mavenExtensions = MavenExtensions.fromFile(extensionsXml);
+
+        assertFalse(mavenExtensions.hasExtension(coordsWithInvalidCharacters));
+    }
+
+    @Test
+    void hasExtensionReturnsFalseWhenOnlyGroupOrArtifactOfCoordinatesArgumentMatches() throws IOException {
+        MavenCoordinates coordsInExtensionsXml = new MavenCoordinates("com.gradle", "some-artifact-id", "0.1.0");
+        MavenCoordinates coordsToMatchSameGroup = new MavenCoordinates("com.gradle", "different-artifact-d");
+        MavenCoordinates coordsToMatchSameArtifact = new MavenCoordinates("com.different", "some-artifact-id");
+        FileUtils.writeLines(extensionsXml, Collections.singleton(generateExtensionsXml(coordsInExtensionsXml)));
+
+        MavenExtensions mavenExtensions = MavenExtensions.fromFile(extensionsXml);
+
+        assertFalse(mavenExtensions.hasExtension(coordsToMatchSameGroup));
+        assertFalse(mavenExtensions.hasExtension(coordsToMatchSameArtifact));
+    }
+
+    @Test
+    void  hasExtensionReturnsTrueWhenGroupAndArtifactOfCoordinatesArgumentMatches() throws IOException {
+        MavenCoordinates coordinatesInExtensionsXml = new MavenCoordinates("com.gradle", "some-artifact-id", "0.1.0");
+        MavenCoordinates coordinatesToMatch = new MavenCoordinates("com.gradle", "some-artifact-id", "0.2.0");
+        FileUtils.writeLines(extensionsXml, Collections.singleton(generateExtensionsXml(coordinatesInExtensionsXml)));
+
+        MavenExtensions mavenExtensions = MavenExtensions.fromFile(extensionsXml);
+
+        assertTrue(mavenExtensions.hasExtension(coordinatesToMatch));
+    }
+
+    @Test
+    void hasExtensionReturnsTrueWhenGroupArtifactAndVersionOfCoordinatesArgumentMatches() throws IOException {
+        MavenCoordinates coordinatesInExtensionsXml = new MavenCoordinates("com.example", "example-artifact-id", "0.1.0");
+        FileUtils.writeLines(extensionsXml, Collections.singleton(generateExtensionsXml(SOME_EXTENSION_COORDINATES, coordinatesInExtensionsXml)));
+
+        MavenExtensions mavenExtensions = MavenExtensions.fromFile(extensionsXml);
+
+        assertTrue(mavenExtensions.hasExtension(SOME_EXTENSION_COORDINATES));
+        assertTrue(mavenExtensions.hasExtension(coordinatesInExtensionsXml));
+    }
```

Comment:
How about an additional test with multiple existing extensions in the xml?

Reply:
The last one in the list should cover that - `hasExtensionReturnsTrueWhenGroupArtifactAndVersionOfCoordinatesArgumentMatches`.
Or did you mean something else?

Reply:
ah right, all good then